tests/saul/main.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ *
+ * @file
+ * @brief       Test the SAUL interface of devices by periodically reading from
+ *              them
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ */
+
+#include <stdio.h>
+
+#include "xtimer.h"
+#include "phydat.h"
+#include "saul_reg.h"
+
+/**
+ * @brief   Read th sensors every second
+ */
+#define INTERVAL            (1000000U)
+
+
+int main(void)
+{
+    phydat_t res;
+    uint32_t last = xtimer_now();
+
+    puts("SAUL test application");
+
+    while (1) {
+        saul_reg_t *dev = saul_reg_get();
```

Comment:
because I don't like these :-)

No, good point, I will look into improving this.